js/lib/AlphabeticIndex.js

```diff
@@ -60,159 +60,291 @@ var Collator = require("./Collator.js");
+ * locale is fully loaded and the address has been parsed. When the onLoad
+ * option is given, the address object
  * will attempt to load any missing locale data using the ilib loader callback.
- * When the constructor is done (even if the data is already preassembled), the 
+ * When the constructor is done (even if the data is already preassembled), the
  * onLoad function is called with the current instance as a parameter, so this
- * callback can be used with preassembled or dynamic loading or a mix of the two. 
- * 
- * <li><i>sync</i> - tell whether to load any missing locale data synchronously or 
+ * callback can be used with preassembled or dynamic loading or a mix of the two.
+ *
+ * <li><i>sync</i> - tell whether to load any missing locale data synchronously or
  * asynchronously. If this option is given as "false", then the "onLoad"
  * callback must be given, as the instance returned from this constructor will
  * not be usable for a while.
  *
- * <li><i>loadParams</i> - an object containing parameters to pass to the 
+ * <li><i>loadParams</i> - an object containing parameters to pass to the
  * loader callback function when locale data is missing. The parameters are not
- * interpretted or modified in any way. They are simply passed along. The object 
+ * interpretted or modified in any way. They are simply passed along. The object
  * may contain any property/value pairs as long as the calling code is in
  * agreement with the loader callback function as to what those parameters mean.
  * </ul>
- * 
+ *
  * @constructor
  * @param {Object} options options to the parser
  */
 var AlphabeticIndex = function (options) {
 	this.sync = true;
 	this.loadParams = {};
-	
+	this.caseSensitive = false;
+	this.accentSensitive = false;
+	this.overflowLabel = "#";
+	this.underflowLabel = "*";
+	this.style = "standard";	
+	this.index = [];
+
 	if (options) {
 		if (options.locale) {
 			this.locale = (typeof(options.locale) === 'string') ? new Locale(options.locale) : options.locale;
 		}
 		
+		if (typeof(options.style) !== 'undefined') {
+			this.style = options.style;
+		}
+
+		if (typeof(options.overflowLabel) !== 'undefined') {
+			this.overflowLabel = options.overflowLabel;
+		}
+
+		if (typeof(options.underflowLabel) !== 'undefined') {
+			this.underflowLabel = options.underflowLabel;
+		}
+		
 		if (typeof(options.sync) !== 'undefined') {
 			this.sync = (options.sync == true);
 		}
-		
 		if (options.loadParams) {
 			this.loadParams = options.loadParams;
 		}
 	}
 
 	this.locale = this.locale || new Locale();
-
-	//console.log("implemented in pure JS");
-	if (!Collator.cache) {
-		Collator.cache = {};
-	}
-
+	
 	Utils.loadData({
-		object: Collator, 
-		locale: this.locale, 
+		object: "AlphabeticIndex",
+		locale: this.locale,
 		name: "collation.json",
-		sync: sync,
-		loadParams: loadParams, 
+		sync: this.sync,
+		loadParams: this.loadParams, 
+
 		callback: ilib.bind(this, function (collation) {
 			if (!collation) {
```

Comment:
We need to be creating a Collator instance here instead of just loading the collation.json data. We will use the collator in a few places in this code anyways. The map can be extracted from the collator instance anyways.